src/Configuration.php

```diff
@@ -486,4 +507,48 @@ public function shouldIgnoreErrorCode($code)
 
         return !(error_reporting() & $code);
     }
+
+    /**
+     * Set session tracking state and pass in optional guzzle.
+     *
+     * @param bool $track whether to track sessions
+     * @param string $endpoint an optional session endpoint
+     *
+     * @return $this
+     */
+    public function setSessionTracking($track, $endpoint = null)
+    {
+        if (!$track) {
+            $this->trackSessions = false;
+
+            return;
+        }
+        if (is_null($endpoint)) {
+            $endpoint = self::SESSION_ENDPOINT;
+        }
+        $this->trackSessions = true;
+        $this->sessionClient = Client::makeGuzzle($endpoint);
```

Comment:
Hmm, not sure I like the config class being able to do this.

Reply:
I agree.  Ideally it would have been better to somehow merge this functionality into the HttpClient, which I may still do.